src/components/SetPasswordForm.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useCallback } from 'react';
+import Form from 'react-bootstrap/Form';
+import '../index.css';
+import '../styles/user/user.css';
+import { Field, Formik } from 'formik';
+import { postAsync } from '../lib/apiRequests/api';
+import { ButtonType, PrimaryButton } from './common/Buttons';
+import { useNavigate, useLocation } from 'react-router-dom';
+
+type SetPasswordFields = {
+  email: string;
+  password: string;
+  repeatPassword: string;
+};
+
+export type SetPasswordRequest = {
+  token: string;
+  email: string;
+  password: string;
+};
+
+export type SetPasswordResponse = {
+  message: string;
+};
+
+const SetPasswordForm: React.FC = () => {
+  const navigate = useNavigate();
+  const queries = useLocation().search;
+  const token = new URLSearchParams(queries).get('token');
+
+  const handleSubmit = useCallback(
+    async (formFields: SetPasswordFields) => {
+      if (token === undefined || token === null) {
+        // TODO: some error page
+        // eslint-disable-next-line no-console
+        console.warn('Tried to reset password without token.');
+        navigate('/login/');
+        return;
+      }
+      await postAsync<SetPasswordResponse, SetPasswordRequest>(
+        '/api/reset-password/set-password/',
+        {
+          token,
+          email: formFields.email,
+          password: formFields.password,
+        }
+      );
+      navigate('/login/');
+    },
+    [navigate, token]
+  );
+
+  return (
+    <div>
+      <h1 className="pb-5">Salasanan palauttaminen</h1>
+      <p>
+        Tilaa salasanan palautuslinkki syöttämällä käyttäjätunnukseesi liitetty
+        sähköpostiosoite.
+      </p>
+      <Formik
+        initialValues={{
+          email: '',
+          password: '',
+          repeatPassword: '',
+        }}
+        validate={() => {
+          // TODO: Add validation checks
+          return {};
+        }}
+        onSubmit={async (values) => handleSubmit(values)}
+      >
+        {(form) => (
+          <Form onSubmit={form.handleSubmit}>
+            <Form.Group className="mb-3">
+              <Form.Label>Sähköpostiosoite</Form.Label>
+              <Field
+                className="form-control"
+                aria-label="Sähköpostiosoite"
+                id="email"
+                name="email"
+                type="email"
+                placeholder=""
+                value={form.values.email}
+                onChange={form.handleChange}
+                autoComplete="on"
+                required
+              ></Field>
```

Comment:
Käytä valmiita komponentteja. Kts `Inputs.tsx`

Reply:
Niissä on valmiina kaikkea kivaa ja tyyli pysyy samana kaikkialla

Reply:
Näin se oli tehty muuallakin :D

Reply:
Niin, loginissa ja registerissä jotka on tehty kauan sitten